cypress/e2e/gui/validarFallBackLoading.cy.js

```diff
@@ -0,0 +1,10 @@
+describe('Validar a mensagem de "Loading"', () => {
+  beforeEach(() => {
+    cy.intercept('GET', '**/customers**').as('getCustomers')
+    cy.visit('/customers?page=1&limit=10&size=All')
+    cy.wait('@getCustomers')
+  })
+  it('Deve exibir a mensagem de carregamento', () => {
```

Comment:
**Este teste é _flaky_!**

Às vezes ele passa e outras falha.

Para testar estados transitórios, como o componente de _loading..._, é importanto forçar o _delay_ na resposta do servidor, conforme explico no blog post [Como simular a demora em uma requisição com Cypress](https://talkingabouttesting.com/2021/02/26/como-simular-a-demora-em-uma-requisicao-com-cypress/).

Veja uma alternantiva para o mesmo teste, porém, a qual provém resultados determinísticos:

```js
describe('Validar a mensagem de "Loading"', () => {
  beforeEach(() => {
    cy.intercept('GET', '**customers**', { delay: 1000 }).as('getDelayedCustomers')
    cy.visit('/')
  })
  it('Deve exibir a mensagem de carregamento', () => {
    cy.get('#loading').should('be.visible')
    cy.wait('@getDelayedCustomers')
    cy.get('#loading').should('not.exist')
  })
})
```

Reply:
Walmyr, eu não consegui entender uma questão, se o meu teste é para que apareça a mensagem de loading...
Porque o cy.wait fica após a validação de `cy.get('#loading').should('be.visible')`, porque colocar após o assert que realmente quero fazer garante que ele não será flaky?
Para mim fez total sentido para a validação de garantir que o loading... não aparece na tela, pois esperei a requisição acabar e então verifiquei que a mensagem não existe.